src/clients/rust/src/lib.rs

```diff
@@ -0,0 +1,844 @@
+    ) -> impl Future<Output = Result<Vec<Transfer>, PacketStatus>> + use<'s> {
+        let (packet, rx) = create_packet::<AccountFilter, tbc::tb_account_filter_t>(
+            tbc::TB_OPERATION_TB_OPERATION_GET_ACCOUNT_TRANSFERS,
+            &[event],
+        );
+
+        unsafe {
+            tbc::tb_client_submit(self.client, Box::into_raw(packet));
+        }
+
+        async {
+            let msg = rx.await.expect("channel");
+            let result: &[tbc::tb_transfer_t] = handle_message(&msg)?;
+
+            Ok(result.iter().cloned().map(Into::into).collect())
+        }
+    }
+
+    pub fn submit_get_account_balances<'s>(
+        &'s self,
+        event: AccountFilter,
+    ) -> impl Future<Output = Result<Vec<AccountBalance>, PacketStatus>> + use<'s> {
+        let (packet, rx) = create_packet::<AccountFilter, tbc::tb_account_filter_t>(
+            tbc::TB_OPERATION_TB_OPERATION_GET_ACCOUNT_BALANCES,
+            &[event],
+        );
+
+        unsafe {
+            tbc::tb_client_submit(self.client, Box::into_raw(packet));
+        }
+
+        async {
+            let msg = rx.await.expect("channel");
+            let result: &[tbc::tb_account_balance_t] = handle_message(&msg)?;
+
+            Ok(result.iter().cloned().map(Into::into).collect())
+        }
+    }
+
+    pub fn submit_query_accounts<'s>(
+        &'s self,
+        event: QueryFilter,
+    ) -> impl Future<Output = Result<Vec<Account>, PacketStatus>> + use<'s> {
+        let (packet, rx) = create_packet::<QueryFilter, tbc::tb_query_filter_t>(
+            tbc::TB_OPERATION_TB_OPERATION_QUERY_ACCOUNTS,
+            &[event],
+        );
+
+        unsafe {
+            tbc::tb_client_submit(self.client, Box::into_raw(packet));
+        }
+
+        async {
+            let msg = rx.await.expect("channel");
+            let result: &[tbc::tb_account_t] = handle_message(&msg)?;
+
+            Ok(result.iter().cloned().map(Into::into).collect())
+        }
+    }
+
+    pub fn submit_query_transfers<'s>(
+        &'s self,
+        event: QueryFilter,
+    ) -> impl Future<Output = Result<Vec<Transfer>, PacketStatus>> + use<'s> {
+        let (packet, rx) = create_packet::<QueryFilter, tbc::tb_query_filter_t>(
+            tbc::TB_OPERATION_TB_OPERATION_QUERY_TRANSFERS,
+            &[event],
+        );
+
+        unsafe {
+            tbc::tb_client_submit(self.client, Box::into_raw(packet));
+        }
+
+        async {
+            let msg = rx.await.expect("channel");
+            let result: &[tbc::tb_transfer_t] = handle_message(&msg)?;
+
+            Ok(result.iter().cloned().map(Into::into).collect())
+        }
+    }
+
+    pub fn close(mut self) -> impl Future<Output = ()> {
+        struct SendClient(tbc::tb_client_t);
+        unsafe impl Send for SendClient {}
+
+        let client = std::mem::replace(&mut self.client, std::ptr::null_mut());
+        let client = SendClient(client);
+
+        let (tx, rx) = channel::<()>();
```

Comment:
I usually use `channel::<std::convert::Infallible>` in situations like this --- the sender being close is already a good wake-up event.

Reply:
Done.